### PR TITLE
CP-12003: optimize "Select a token" modal

### PR DESCRIPTION
**Pull request by atn4z7**
## Description

**Ticket: [CP-CP-12003]**

Please provide:

- A summary of the changes
- Motivation and context for this change
- Any dependencies introduced
- (If breaking) migration steps and instructions

## Screenshots/Videos

Include relevant screenshots or screen recordings of iOS and Android.

## Testing

**Dev Testing (if applicable)**

- Provide steps to test the happy path of your feature
- Provide steps to test edge cases and error states
- Trigger a build on bitrise and reference it here
- Move the ticket into the "Testing" column on Jira

**QA Testing (if applicable)**

- Provide instructions for QA to test this feature thoroughly
- State expected behavior / acceptance criteria

## Checklist

Please check all that apply (if applicable)

- [ ] I have performed a self-review of my code
- [ ] I have verified the code works
- [ ] I have included screenshots / videos of android and ios
- [ ] I have added testing steps
- [ ] I have added/updated necessary unit tests
- [ ] I have updated the documentation

### CODE REVIEW

File: packages/core-mobile/app/new/features/meld/store.ts [hunk 1]
```diff
@@ -57,3 +62,105 @@ export const useOfframpActivityIndicator = (): {
...
+type SupportedCryptoCurrenciesStore = {
+  supportedCryptoCurrencies: CryptoCurrencyWithBalance[]
+  setSupportedCryptoCurrencies: (
+    cryptos: CryptoCurrency[],
+    tokenIndexes: TokenIndex
+  ) => void
+  reset: () => void
+}
+
+const supportedCryptoCurrenciesStore = create<SupportedCryptoCurrenciesStore>(
```
Comment (atn4z7) [inline packages/core-mobile/app/new/features/meld/store.ts:117]: also caching the data in memory/zustand store so that every time user opens the select a token screen, it doesn't recompute. this data will get reset whenever the buy/sell flow starts so a new list can get computed.
